### PR TITLE
Wp 926 cleanup

### PR DESCRIPTION
**Pull request by xxbillouxx**
Split code into functions for better reading

### CODE REVIEW

**Comment by xxbillouxx** [comment]
recheck

**Comment by DrPyser** [comment]
recheck

**Comment by DrPyser** [comment]
I'm getting errors when running tests locally, weirdly not reproduced in CI.

**Comment by DrPyser** [comment]
> I'm getting errors when running tests locally, weirdly not reproduced in CI.

Some local docker caching issue, fixed it and tests are passing fine.

**Comment by DrPyser** [comment]
> I'm getting errors when running tests locally, weirdly not reproduced in CI.

Some local docker caching issue, fixed it and tests are passing fine.

File: xivo_dao/resources/utils/search.py [hunk 1]
```diff
@@ -194,3 +175,29 @@ def _paginate(self, query, limit=None, offset=0):
...
+        limit = parameters.pop('limit', None)
+        offset = parameters.pop('offset', 0)
+        reverse = False if parameters.pop('direction', 'asc') == 'asc' else True
+        return parameters, order, limit, offset, reverse
+
+    def _apply_search_params(self, rows, order, limit, offset, reverse):
+        if order:
+            rows = sorted(
+                rows,
+                key=lambda x: unicodedata.normalize('NFKD', x.__getattribute__(order)),
```
Comment (DrPyser) [inline xivo_dao/resources/utils/search.py:194]:
```suggestion
                key=lambda x: unicodedata.normalize('NFKD', getattr(x, "order")),
```
getattr is the public interface for dynamic attribute access.

**Review by DrPyser — APPROVED** [review]

**Comment by DrPyser** [comment]
Good refactoring, thumbs up

**Pull request merged**